packages/@eventual/aws-cdk/src/entity-service.ts

```diff
@@ -239,6 +239,24 @@ class Entity extends Construct {
             ? StreamViewType.NEW_AND_OLD_IMAGES
             : StreamViewType.NEW_IMAGE
           : undefined,
+    }));
+
+    props.entity.indices.forEach((i) => {
+      // LSI may not be used with a table that doesn't already have a sort key, but GSI can
+      if (i.partition || !keyDefinition.sort) {
+        table.addGlobalSecondaryIndex({
+          indexName: i.name,
+          partitionKey: entityKeyDefinitionToAttribute(i.key.partition),
+          sortKey: i.key.sort
+            ? entityKeyDefinitionToAttribute(i.key.sort)
+            : undefined,
+        });
+      } else if (i.key.sort) {
```

Comment:
What if the partition is the same and only differs on sort? Should that be a LSI?

Reply:
I thought about that..., But left the ability to explicitly create a gsi when the partition is given explicitly. Another option would be to use a flag to do that. This thoughts?

This would only matter when the original table has a sort key.